Fix broken build when TRACE logging is enabled

`src/mjolnir/graphenhancer.cc` is missing a header for one of the trace logging statements:
```
In file included from /home/mandeep/work/valhalla/valhalla/baldr/graphtileheader.h:9,
                 from /home/mandeep/work/valhalla/valhalla/baldr/graphtile.h:13,
                 from /home/mandeep/work/valhalla/valhalla/mjolnir/graphtilebuilder.h:18,
                 from /home/mandeep/work/valhalla/valhalla/mjolnir/admin.h:15,
                 from /home/mandeep/work/valhalla/src/mjolnir/graphenhancer.cc:2:
/home/mandeep/work/valhalla/src/mjolnir/graphenhancer.cc: In function ‘void {anonymous}::enhance(const ptree&, const valhalla::mjolnir::OSMData&, const string&, const ptree&, std::queue<valhalla::baldr::GraphId>&, std::mutex&, std::promise<{anonymous}::enhancer_stats>&)’:
/home/mandeep/work/valhalla/src/mjolnir/graphenhancer.cc:1852:23: error: ‘format’ is not a member of ‘boost’; did you mean ‘date::format’?
 1852 |     LOG_TRACE((boost::format("GraphEnhancer completed tile %1%") % tile_id).str());
      |                       ^~~~~~
```

This should be fixable by including `boost/format.hpp`.

--- a/src/mjolnir/graphenhancer.cc
+++ b/src/mjolnir/graphenhancer.cc
@@ -18,6 +18,7 @@
 #include <unordered_set>
 #include <vector>
 
+#include <boost/format.hpp>
 #include <boost/geometry.hpp>
 #include <boost/geometry/geometries/point_xy.hpp>
 #include <boost/geometry/geometries/polygon.hpp>

